objax/optimizer/lars.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+__all__ = ['LARS']
+
+from typing import List
+
+import jax.numpy as jn
+
+from objax.module import Module, ModuleList
+from objax.typing import JaxArray
+from objax.variable import TrainRef, StateVar, TrainVar, VarCollection
+
+
+class LARS(Module):
+    """Layerwise adaptive rate scaling (LARS) optimizer.
+
+    See https://arxiv.org/abs/1708.03888
+    """
+
+    def __init__(self, vc: VarCollection,
+                 momentum: float = 0.9,
+                 beta: float = 1e-4,
```

Comment:
Let's rename this to `weight_decay` since `beta` sounds obscure.

Reply:
I was using beta because the paper calls it beta but I switched to weight_decay.